client/my-sites/email/email-management/home/email-plan-warning-notice.tsx

```diff
@@ -20,6 +21,12 @@ export const EmailPlanWarningNotice = ( props: EmailPlanWarningNoticeProps ) =>
 	const cannotAddEmailWarningMessage = cannotAddEmailWarningReason?.message ?? '';
 	const cannotAddEmailWarningCode = cannotAddEmailWarningReason?.code ?? null;
 
+	// If email and domain are owned by different users, none of the users will be able to make a purchase and the only way to resolve this
+	// is to reach out to support. Therefore, we should surface a different message to address this scenario.
+	if ( ! canDomainOwnerAddEmail( domain ) ) {
+		return <EmailDifferentDomainOwnerMessage source="email-management" />;
+	}
+
```

Comment:
We could move that `if` clause to the top of this method - there is no need to retrieve and destructure the email reason if the domain owner cannot add any email anyway.